includes/class-sensei-course-progress-widget.php

```diff
@@ -130,8 +131,38 @@ function widget( $args, $instance ) {
 		<?php endif; ?>
 
 		<?php
+
+		$display_modules_in_lesson_sidebar = get_post_meta( $lesson_course_id, 'cxl_display_modules_in_lesson_sidebar', true );
+		$old_module                        = false;
+
 		foreach( $lesson_array as $post ) {
 
+			/*
+			 * Display module titles upon choice set in course meta.
+			 * Used part of the original implementation that is in master branch.
+			 *
+			 * @since 2024.08.29
+			 * @see https://github.com/woocommerce/sensei-course-progress/blob/150c4472acf3555eb5f870fd17daa25fe311a8f3/includes/class-sensei-course-progress-widget.php#L247
+			 */
+			if ( $display_modules_in_lesson_sidebar && isset( Sensei()->modules ) ) {
+				$new_module = Sensei()->modules->get_lesson_module( $post->ID );
+
+				if ( $old_module != $new_module ) {
+					if ( $new_module ) {
+						$module_title = $this->get_module_title_content( $new_module );
+					} else {
+						$module_title = esc_html( __( 'Other Lessons', 'sensei-course-progress' ) );
+					}
+
+					?>
+					<h5 class="cxl-module-title">
```

Comment:
I'd simplify it.

```suggestion
					<h5 class="module-title">
```

Reply:
Indeed, I usually try to practice a `cxl` prefix so that later when someone reads the template, its clear that "this part is not upstream".

But I'm cool with either way.

Reply:
I looked at the original solution to linked to and it had `module-title` class so I consider it upsteam